### PR TITLE
Remove the index type from internal watcher indexes

### PR DESCRIPTION
**Pull request by jakelandis**
This commit removes the "doc" type from watcher internal indexes.
The template still carries the "_doc" type since that is needed for
the internal representation.

This impacts the .watches, .triggered-watches, and .watch-history indexes.

External consumers do not need any changes since all external calls
go through the _watcher API, and should not interact with the the .index directly.

Relates #38637

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
     public static final String TYPE = "index";
 
-    @Nullable final String docType;
```
Comment (jakelandis) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/actions/index/IndexAction.java]: The changes to this file (IndexAction) were missed in https://github.com/elastic/elasticsearch/pull/37594. I added it here to help us find what to remove when we completely get rid of types. Apologizes for conflating the two issues.

**Review by jpountz — APPROVED** [review]
LGTM

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/actions/ack/TransportAckWatchActionTests.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void testWatchNotFound() {
         String watchId = "my_watch_id";
         doAnswer(invocation -> {
             ActionListener<GetResponse> listener = (ActionListener<GetResponse>) invocation.getArguments()[1];
-            listener.onResponse(new GetResponse(new GetResult(Watch.INDEX, Watch.DOC_TYPE, watchId, UNASSIGNED_SEQ_NO, 0, -1, false,
+            listener.onResponse(new GetResponse(new GetResult(Watch.INDEX, null, watchId, UNASSIGNED_SEQ_NO, 0, -1, false,
```
[the remark this answers is not in the document]
Comment (hub-cap) [reply x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/actions/ack/TransportAckWatchActionTests.java]: ++ to changing that to use default mapping instead

**Review by hub-cap — APPROVED** [review]
Pending just the one nit that @jpountz mentioned, im also approving this.

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**